src/os/bluestore/BlueStore.cc

```diff
@@ -8787,6 +8787,8 @@ void BlueStore::_kv_finalize_thread()
       }
 
       // this is as good a place as any ...
+      // empty don't be protected by reap_lock.It can do next time.
+      if (!removed_collections.empty())
```

Comment:
I am afraid acquiring the **reap_lock** here is inevitable since we cannot access removed_collections safely without the protection of it.  Might fold this check into `_reap_collections` instead?

Reply:
Although access isn't safe.  If we omit this we can do next time. I think remove_collection don't care this latency.

Reply:
From the code, _txc_finish queue removed_collection. And _reap_collection in the same thread. So we don't care omit this.

Reply:
Can you fix the indention below?